external_parser/parse_example_external.cc

```diff
@@ -94,18 +62,19 @@ void apply_cli_overrides(std::unique_ptr<i_joiner>& joiner, vw *all, const input
   }
   if(all->options->was_supplied("learning_mode")) {
     v2::LearningModeType learning_mode;
-    if(!str_to_learning_mode(parsed_options.ext_opts->learning_mode, learning_mode)) {
+    if(!str_to_enum(parsed_options.ext_opts->learning_mode, learning_modes, v2::LearningModeType_MIN, learning_mode)) {
       throw std::runtime_error("Invalid argument to --problem_type " + parsed_options.ext_opts->learning_mode);
     }
     joiner->set_learning_mode_config(learning_mode, true);
   }
   if(all->options->was_supplied("reward_function")) {
     v2::RewardFunctionType reward_function;
-    if(!str_to_reward_function(parsed_options.ext_opts->reward_function, reward_function)) {
+    if(!str_to_enum(parsed_options.ext_opts->reward_function, reward_functions, v2::RewardFunctionType_MIN, reward_function)) {
       throw std::runtime_error("Invalid argument to --problem_type " + parsed_options.ext_opts->reward_function);
     }
     joiner->set_reward_function(reward_function, true);
   }
+  joiner->apply_cli_overrides(all, parsed_options);
```

Comment:
is this something that the `joiner` needs to know about? can't it be handled at this level?

Reply:
This PR is about adding multistep specific parameter "multistep_reward" which is controlling behavior specific to multistep_joiner. I agree that it does not look great, but not sure about other ways to generalize this logic across multistep/non multistep joiner implementations.

Reply:
ok maybe we could create a multistep `parse_example_multistep` like we have a `parse_example_converter` and rename `parse_example_external` to something else, and have the options available there so that each parser can set it's own options

might be outside of the scope of this PR

Reply:
yes I am going to schedule some refactoring/ cleaning up work for myself and will try to address it  there.